### PR TITLE
fix(typescript): fix type errors in converted code

### PR DESCRIPTION
**Pull request by zxbodya**
part of https://github.com/uber/baseweb/pull/4989, depends on https://github.com/uber/baseweb/pull/5008

#### Description

Updating converted code fixing type errors in it.
This is to contain most of the actual changes for the migration to review.

### CODE REVIEW

**Review by chasestarr — APPROVED** [review]

File: src/menu/maybe-child-menu.tsx [hunk 1]
```diff
@@ -8,31 +8,32 @@ LICENSE file in the root directory of this source tree.
...
 
 import { NestedMenuContext } from './nested-menus';
-import { Popover } from '../popover/index';
+import { Popover } from '../popover';
 import type { OverrideT } from '../helpers/overrides';
 import { getOverrides, mergeOverrides } from '../helpers/overrides';
 import type { ItemT } from './types';
 
 type PropsT = {
-  children: React.ReactNode;
```
Comment (chasestarr) [inline src/menu/maybe-child-menu.tsx:17]: Does React.FC imply `children` prop? Is removing this the reason for ts-expect-errors comments below?

File: src/button-group/types.ts [hunk 2]
```diff
@@ -84,4 +84,6 @@ export type StateReducerT = (
 ) => StateT;
 
 // general
-type ClickHandlerT = (event: SyntheticEvent<HTMLButtonElement>, index: number) => unknown;
+type ClickHandlerT = (...args: any[]) => void;
+// todo: causes type errors, and does not look right
+// type ClickHandlerT = (event: SyntheticEvent<HTMLButtonElement>, index: number) => unknown;
```
Comment (chasestarr) [inline src/button-group/types.ts:89]: should be correct based on usage here https://github.com/uber/baseweb/pull/5009/files#diff-3a7f2280f8ed9c3b284d87b0a18297b198b4f35f4944334fbf2de980b78e7a8dR118 What is the error?

File: src/styles/util.ts [hunk 3]
```diff
@@ -25,22 +26,10 @@ export const ellipsisText = {
...
-  borderBottomStyle: Globals | LineStyle;
-  borderBottomColor: string;
-  borderLeftWidth: string;
-  borderLeftStyle: Globals | LineStyle;
-  borderLeftColor: string;
-  borderRightWidth: string;
-  borderRightStyle: Globals | LineStyle;
-  borderRightColor: string;
-} {
+// todo(flow->ts) backport
```
[the remark this answers is not in the document]
Comment (zxbodya) [reply src/styles/util.ts:31]: was adding it where "fix" can be ported back to flow sources
(there was some fixes, I was creating PRs earlier, from such fixes)

or this one - it only makes type annotation smaller, so it does not really need to be copied to flow -- will remove the comment

File: src/radio/types.ts [hunk 4]
```diff
@@ -197,7 +183,7 @@ export type StatefulRadioGroupPropsT = {
 };
 
 export type StylePropsT = {
-  $align?: boolean;
+  $align?: string; // todo: incorrect flow type - 'horizontal' is expected
```
[the remark this answers is not in the document]
Comment (zxbodya) [reply src/radio/types.ts:186]: it probably can be further improved restricting it to 'horizontal'|'vertical' (the actual value here looks to be limited to `ALIGN` constant)

updating todo with a clarification

File: src/menu/nested-menus.tsx [hunk 5]
```diff
@@ -18,6 +18,7 @@ type PropsT = {
   children: React.ReactNode;
 };
 
+// @ts-expect-error todo: incorrect default value
```
[the remark this answers is not in the document]
Comment (zxbodya) [reply src/menu/nested-menus.tsx:21]: yes, when calling `createContext` explicit type argument it infers it from default value passed into the call and it is different from `NestedMenuContextT`, also here default `getParrentMenu` and `getChildMenu` looks to have incorrect return values

updating the fix to use type argument, and adding ts-ignores to specific properties not matching the types

File: src/menu/maybe-child-menu.tsx (hunk 1 above)
Comment (zxbodya) [reply src/menu/maybe-child-menu.tsx:17]: yes, `React.FC` will anyway add it with the same type, so it is not really needed here (adding it back btw will not change anything)

about ts-expect-errors below - that is unrelated… while technically react does allow `ReactNode` to be returned from functional components, typescript does not (there are some reasons about type-checking JSX), and that is why ts-expect-errors in code below

File: src/datepicker/types.ts [hunk 6]
```diff
@@ -340,49 +339,51 @@ export type StatefulDatepickerPropsT<PropsT, T = Date> = Omit<
...
-  /**
-   * Optional value that can be provided to fully control the component. If not provided,
-   * TimezonePicker will manage state internally.
-   */
-  value?: string | null;
-  disabled?: boolean;
-  error?: boolean;
-  positive?: boolean;
-};
+// todo: unused duplicated types
```
[the remark this answers is not in the document]
Comment (zxbodya) [reply src/datepicker/types.ts:342]: removing

File: src/popover/types.ts [hunk 7]
```diff
@@ -72,19 +65,19 @@ export type BasePopoverPropsT = {
   /** Where to mount the popover */
   mountNode?: HTMLElement;
   /** Handler for blur events on trigger element. */
-  onBlur?: (e: Event) => unknown;
+  onBlur?: (e: React.FocusEvent) => unknown;
   /** Handler for click events on trigger element. */
-  onClick?: (e: Event) => unknown;
+  onClick?: (e: React.SyntheticEvent) => unknown;
```
[the remark this answers is not in the document]
Comment (zxbodya) [reply src/popover/types.ts:70]: missed the comment - indeed it should be MouseEvent, also looking into it - there are other small improvements to be made here - will push a fix

**Pull request merged**